traffic_ops/app/lib/API/User.pm

```diff
@@ -281,11 +284,14 @@ sub current {
 
 	if ( &is_ldap($self) ) {
 		my $role = $self->db->resultset('Role')->search( { name => "read-only" } )->get_column('id')->single;
+		my $user_tenant_id = $self->current_user_tenant();
+		my $user_tenant = defined($user_tenant_id) ? $self->db->resultset('Tenant')->search( { id => $user_tenant_id } )->get_column('name')->single : "N/A";
```

Comment:
it will work without prefetch....but it will create N+1 queries. For example, if you do a query of the TMUser table which returns 100 users and then you loop thru your results and call $row->tenant->name, it will do 1 query for each user to fetch the tenant so you end up with 101 queries.

When you use prefetch you join the user table to the tenant table in the first query so you only end up with 1 query...

1 query is better than 101 :)